gateway-provider-ha/src/main/java/org/apache/knox/gateway/ha/provider/impl/HaDescriptorFactory.java

```diff
@@ -31,19 +31,81 @@ public static HaDescriptor createDescriptor() {
 
    public static HaServiceConfig createServiceConfig(String serviceName, String config) {
       Map<String, String> configMap = parseHaConfiguration(config);
-      String enabledValue = configMap.get(CONFIG_PARAM_ENABLED);
-      String maxFailoverAttempts = configMap.get(CONFIG_PARAM_MAX_FAILOVER_ATTEMPTS);
-      String failoverSleep = configMap.get(CONFIG_PARAM_FAILOVER_SLEEP);
-      String zookeeperEnsemble = configMap.get(CONFIG_PARAM_ZOOKEEPER_ENSEMBLE);
-      String zookeeperNamespace = configMap.get(CONFIG_PARAM_ZOOKEEPER_NAMESPACE);
-      String stickySessionEnabled = configMap.get(CONFIG_STICKY_SESSIONS_ENABLED);
-      String loadBalancingEnabled = configMap.get(CONFIG_LOAD_BALANCING_ENABLED);
-      String stickySessionCookieName = configMap.get(STICKY_SESSION_COOKIE_NAME);
-      String noFallbackEnabled = configMap.get(CONFIG_NO_FALLBACK_ENABLED);
-      return createServiceConfig(serviceName, enabledValue, maxFailoverAttempts, failoverSleep,
-          zookeeperEnsemble, zookeeperNamespace, loadBalancingEnabled, stickySessionEnabled, stickySessionCookieName, noFallbackEnabled);
+      boolean enabled = DEFAULT_ENABLED;
+      int maxFailoverAttempts = DEFAULT_MAX_FAILOVER_ATTEMPTS;
+      int failoverSleep = DEFAULT_FAILOVER_SLEEP;
+      boolean stickySessionsEnabled = DEFAULT_STICKY_SESSIONS_ENABLED;
+      boolean loadBalancingEnabled = DEFAULT_LOAD_BALANCING_ENABLED;
+      boolean noFallbackEnabled = DEFAULT_NO_FALLBACK_ENABLED;
+      String stickySessionCookieName = DEFAULT_STICKY_SESSION_COOKIE_NAME;
+      String disableStickySessionForUserAgents = DEFAULT_DISABLE_LB_USER_AGENTS;
+
+     String enabledValue = configMap.get(CONFIG_PARAM_ENABLED);
+     String maxFailoverAttemptsValue = configMap.get(CONFIG_PARAM_MAX_FAILOVER_ATTEMPTS);
+     String failoverSleepValue = configMap.get(CONFIG_PARAM_FAILOVER_SLEEP);
```

Comment:
You may use `Map.getOrDefault(...)` to save the above default settings.

Even better, if I were you I'd add some private methods like that:
```
getBooleanValue(Map configMap, String configName, boolean default) {
  final configValue = configMap.get(configName);
  return StringUtils.isNotBlank(configValue) ? Boolean.parseBoolean(configValue) : default;
}
```
Then you could initialize your local variables like this:
```
final boolean enabled = getBooleanValue(configMap, CONFIG_PARAM_ENABLED, DEFAULT_ENABLED);
final int maxFailoverAttempts = getIntValue(configMap, CONFIG_PARAM_MAX_FAILOVER_ATTEMPTS, DEFAULT_MAX_FAILOVER_ATTEMPTS);
...
```